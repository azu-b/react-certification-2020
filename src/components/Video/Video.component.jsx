import React from 'react';
import { Container, VideoContainer, Title, Description } from './Video.styled';

const Video = (props) => {
  const { id, title, description } = props;

  return (
    <Container>
      <VideoContainer src={`https://www.youtube.com/embed/${id}`} />
      <Title>{title}</Title>
      <Description>{description}</Description>
    </Container>
  );
};

export default Video;
