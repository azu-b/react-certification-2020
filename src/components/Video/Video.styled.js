import styled from 'styled-components';

const Container = styled.div`
  display: flex;
  flex-direction: column;
`;

const VideoContainer = styled.iframe`
  width: 100%;
  height: 250px;

  @media (min-width: 768px) {
    height: 500px;
  }

  @media (min-width: 1440px) {
    height: 600px;
  }
`;

const Title = styled.h1`
  margin: 10px;
  font-family: 'Folks Normal';
  font-size: 36px;
  color: #330080;
`;

const Description = styled.p`
  margin: 10px;
  font-family: 'Heebo Light';
  color: grey;
  max-height: 200px;
  overflow-y: scroll;
  white-space: pre-wrap;
`;

const Error = styled.h2`
  font-family: 'Folks Normal';
  color: #9c0000;
  padding: 10px;
`;

const FavoriteButton = styled.button`
  font-size: 16px;
  font-family: 'Heebo Medium';
  background-color: ${(props) => (props.add ? '#00a541' : '#e27800')};
  color: white;
  border: none;
  padding: 10px 20px;
  cursor: pointer;
`;

export { Container, VideoContainer, Title, Description, Error, FavoriteButton };
