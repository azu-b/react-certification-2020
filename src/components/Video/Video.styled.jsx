import styled from 'styled-components';

const Container = styled.div`
  display: flex;
  flex-direction: column;
`;

const VideoContainer = styled.iframe`
  width: 100%;
  height: 500px;
`;

const Title = styled.h1`
  color: black;
`;

const Description = styled.p`
  color: grey;
`;

export { Container, VideoContainer, Title, Description };
