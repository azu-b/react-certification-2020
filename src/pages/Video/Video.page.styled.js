import styled from 'styled-components';

const Container = styled.div`
  margin-top: 60px;
  display: flex;
  flex-direction: column;

  @media (min-width: 1024px) {
    flex-direction: row;
  }
`;

const VideoContainer = styled.div`
  width: 100%;
  padding: 10px;

  @media (min-width: 1024px) {
    width: 60%;
    padding: 20px;
  }
`;

const RelatedVideosContainer = styled.div`
  width: 100%;
  padding: 10px;

  @media (min-width: 1024px) {
    width: 40%;
    padding: 20px;
  }
`;

const Error = styled.div`
  font-family: 'Folks Normal';
  color: #9c0000;
  font-size: 32px;
  margin: 20px;

  @media (min-width: 768px) {
    font-size: 36px;
    margin: 20px auto;
  }
`;

const Load = styled.div`
  font-family: 'Heebo Regular';
  color: #0093ff;
  font-size: 28px;
  margin: 20px;

  @media (min-width: 768px) {
    font-size: 32px;
    margin: 20px auto;
  }
`;

const LoadRelated = styled(Load)`
  font-size: 20px;
  margin: 10px;

  @media (min-width: 768px) {
    font-size: 24px;
    margin: 10px auto;
  }
`;

export { Container, VideoContainer, RelatedVideosContainer, Error, Load, LoadRelated };
