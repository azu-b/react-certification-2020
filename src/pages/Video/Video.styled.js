import styled from 'styled-components';

const Container = styled.div`
  margin-top: 60px;
  display: flex;
  justify-content: center;
  align-items: flex-start;
`;

export { Container };
