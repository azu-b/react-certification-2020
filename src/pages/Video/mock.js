const mockVideo = {
  id: 'uE-1RPDqJAY',
  description: 'THEY"RE TAKING THE HOBBITS TO ISENGARD (not my work)',
  title: 'taking the hobbits to isengard',
};

export default mockVideo;
