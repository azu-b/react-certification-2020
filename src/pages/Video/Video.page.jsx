import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import VideoComponent from '../../components/Video';
import VideoList from '../../components/VideoList';
import { Container, VideoContainer, RelatedVideosContainer } from './Video.styled';
import { YouTubeAPI, cleanYouTubeResponse } from '../../utils/youtube';

const Video = () => {
  const { id: videoId } = useParams();
  const [video, setVideo] = useState(undefined);
  const [relatedVideos, setRelatedVideos] = useState(undefined);
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    const getVideoInfo = async () => {
      try {
        const response = await YouTubeAPI.get('/videos', {
          params: {
            id: videoId,
          },
        });
        const cleanVideos = cleanYouTubeResponse(response.data.items);
        console.log(cleanVideos);
        setVideo(cleanVideos[0]);
        setIsLoading(false);
      } catch (error) {
        console.log(error);
      }
    };

    const getRelatedVideos = async () => {
      try {
        const response = await YouTubeAPI.get('/search', {
          params: {
            relatedToVideoId: videoId,
            type: 'video',
            maxResults: 25,
          },
        });
        const cleanVideos = cleanYouTubeResponse(response.data.items);
        setRelatedVideos(cleanVideos);
      } catch (error) {
        console.log(error);
      }
    };

    getVideoInfo();
    getRelatedVideos();
  }, [videoId]);

  return (
    <Container>
      {isLoading && <div>Loading video info...</div>}
      {video ? (
        <VideoContainer>
          <VideoComponent
            id={videoId}
            title={video.title}
            description={video.description}
          />
        </VideoContainer>
      ) : (
        <div>Ups! No video with this id found</div>
      )}
      {relatedVideos && (
        <RelatedVideosContainer>
          <VideoList videos={relatedVideos} />
        </RelatedVideosContainer>
      )}
    </Container>
  );
};

export default Video;
