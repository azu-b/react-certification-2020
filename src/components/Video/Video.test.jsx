import React from 'react';
import { render } from '@testing-library/react';
import Video from './index';

describe('<Video>', () => {
  const props = {
    id: 'uE-1RPDqJAY',
    title: 'taking the hobbits to isengard',
    description: `THEY"RE TAKING THE HOBBITS TO ISENGARD

      (not my work)`,
  };

  it('renders', () => {
    const { getByTestId } = render(<Video {...props} />);
    const video = getByTestId('video');
    const title = getByTestId('title');
    const description = getByTestId('description');

    expect(video.tagName).toBe('IFRAME');
    expect(title.tagName).toBe('H1');
    expect(description.tagName).toBe('P');
  });

  it('matches snapshot', () => {
    const { container } = render(<Video {...props} />);
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('renders a message when no props are provided', () => {
    const { container } = render(<Video />);
    expect(container.innerHTML).toMatch('No video information was provided');
  });
});
