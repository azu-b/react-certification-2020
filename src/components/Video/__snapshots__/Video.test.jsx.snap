// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Video> matches snapshot 1`] = `
"<div class=\\"sc-bdnylx jiVbiK\\"><iframe data-testid=\\"video\\" src=\\"https://www.youtube.com/embed/uE-1RPDqJAY\\" class=\\"sc-gtssRu bTjPez\\"></iframe>
  <h1 data-testid=\\"title\\" class=\\"sc-dlnjPT cuonUe\\">taking the hobbits to isengard</h1>
  <p data-testid=\\"description\\" class=\\"sc-hKFyIo ckbdfc\\">THEY\\"RE TAKING THE HOBBITS TO ISENGARD

    (not my work)</p>
</div>"
`;
